clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -956,8 +956,16 @@ PointerVariableConstraint::mkString(Constraints &CS,
   }
 
   // No space after itype.
-  if (!EmittedName && !UseName.empty())
-    Ss << " " << UseName;
+  if (!EmittedName && !UseName.empty()) {
+    // Avoid emitting an extra space between the last star and the identifier
+    // when generating strings for unchecked pointers. This is used when
+    // emitting the unchecked part of typedef parameters with itypes. Ss.str()
+    // unfortunately copies contents of Ss. With C++20 there will be Ss.view()
+    // which I believe avoids the copy.
+    if (!Ss.str().empty() && Ss.str().back() != '*')
+      Ss << " ";
+    Ss << UseName;
```

Comment:
Heh, I have [an equivalent change](https://github.com/correctcomputation/checkedc-clang/pull/657/files#diff-af8e5524b0a2a27e2e1345d252c135b9f228b3c3f6aa789fca9f76b2cca1e725R979-R986) in #657.  Assuming this PR goes in first, you can copy the rest of the updates to the expected output of regression tests from #657.